pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -114,6 +121,51 @@ func podKeyGet(pod *corev1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
+// buildVFDeviceIDListPerPod is a helper function to build a cache structure with the
+// list of all the PCI addresses allocated per Pod based on their resource requests (in Pod spec).
+// When there is a request for a VF resource (to associate it for a secondary network interface),
+// getUnusedSriovVFDeviceIDPerPod will use this cache information to pick up a unique PCI address
+// which is still not associated with a network device name.
+// NOTE: buildVFDeviceIDListPerPod is called only if a Pod specific VF to Interface mapping cache
+// was not build earlier. Sample initial entry per Pod: "{18:01.1,""},{18:01.2,""},{18:01.3,""}"
+func (pc *PodController) buildVFDeviceIDListPerPod(podName, podNamespace string) error {
+	key := podNamespace + "/" + podName
+	if pc.vfDeviceIDUsageMap[key] != nil {
+		return nil
+	}
+	podSriovVFDeviceIDs, err := getPodContainerDeviceIDs(podName, podNamespace)
+	if err != nil {
+		return fmt.Errorf("getPodContainerDeviceIDs failed: %v", err)
+	}
+	var vfDeviceIDInfo []podContainerDeviceIDInfo
+	for _, pciAddress := range podSriovVFDeviceIDs {
+		initVfDeviceID := podContainerDeviceIDInfo{vfDeviceID: pciAddress, ifName: ""}
+		vfDeviceIDInfo = append(vfDeviceIDInfo, initVfDeviceID)
+		pc.vfDeviceIDUsageMap[key] = vfDeviceIDInfo
+	}
+	return nil
+}
+
+func (pc *PodController) getUnusedSriovVFDeviceIDPerPod(podName, podNamespace, interfaceName string) (string, error) {
```

Comment:
Should we change "get" to "allocate" or "assign", like allocateSriovVFDeviceIDForInterface? As the func called "get", I did not understand it actually assign the VF to the interface and change the cache data.